ruby/examples/lib/apply_order_discount.rb

```diff
@@ -0,0 +1,44 @@
+class ApplyOrderDiscount
+  Request = Struct.new(:id)
+  Response = Struct.new(:discount, :has_discount?)
+
+  attr_reader :gateway
+
+  def initialize(gateway: Gateway.new)
```

Comment:
Referencing the gateway here puts a source-code dependency on Gateway, which is undesirable since it couples this class to the Gateway which creates immobility